### DISCUSSION TITLE
List all repos(internal included) from a Github org using API

### DISCUSSION THREAD

**Discussion by ydorbala** [community/community#24769] [Apps, API and Webhooks]
<p>I am trying to list all the repos(public, internal) within the org that I am the owner of using the below command.</p>
<p><code>$orgRepos = Invoke-WebRequest -H $headers -Uri "https://api.github.com/orgs/{$orgName}/repos"</code></p>
<p>There are about 2 public repos and 10+ internal repos but the above command is listing out <strong>only</strong> the 2 public repos. I went through the documentation and tried all the options as suggested <a href="https://docs.github.com/en/rest/reference/repos" rel="noopener nofollow ugc">here</a> with <code>type</code> and <code>per_page</code> but nothing seems to help. For headers, I created it using my PAT by giving access to all the permissions.</p>
<pre><code class="lang-auto">$headers = @{
      Authorization = "Basic $($token)"
      Accept = "application/vnd.github.v3+json"
}
</code></pre>
<p>How do I get to list out the internal repositories?</p>

**Comment by mickeygousset** [comment]
<p>Are you using GHEC or GHES?</p>

**Comment by ydorbala** [comment]
<p>We use GHEC. Thanks!</p>

**Comment by mickeygousset** [comment]
<p>I can’t recreate this. When I run the same command against my GHEC, it is returning all the repos.  Interesting, it shows internal as private, but otherwise it is returning all the repos.</p>
<p>Check your PAT. Does it have the appropriate rights? Can you create a token with all rights, just to make sure?  Are you using SSO? Did you click the enable sso on your PAT?</p>

**Comment by ydorbala** [comment]
<p>Thanks for looking into this. Just for confirmation, you had run the same command that I had listed above and its listing out all the internal repos as well?</p>
<p>Yes, as for the PAT, I went ahead and created one with all the permissions but am still not able to see the internal repos. One of the other team members with similar admin permissions is seeing the same issue as well. We use SSO and enabled it for the PAT too.</p>
<p>Thanks!</p>

**Comment by mickeygousset** [comment]
<p>oh wait. My curl command is slightly differnet :slight_smile:</p>
<p>If I use “basic”, I don’t get back my internal ones.</p>
<p>However, if I use Token, I do:</p>
<p>curl -H ‘Authorization: token your-token-here’ ‘<a href="https://api.github.com/orgs/your">https://api.github.com/orgs/your</a> org here/repos’</p>

**Comment by mickeygousset** [comment, marked as answer]
<p>In your header, replace “Basic” with “token” and see what happens.</p>

**Comment by ydorbala** [comment]
<p>Awesome, that worked, thanks! would be great if the documentation covers this as well :slight_smile: thanks for your time again!</p>

**Comment by huttsMichael** [comment]
Important note for anyone experiencing this issue. It seems, at least from my testing, that fine-grained tokens do not work for this. A classic token of similar/identical permissions should work just fine with this answer though. Hope this helps anyone else racking their brain about why this solution doesn't seem to work :)
